### DISCUSSION TITLE
"import" a local conda environment?

### DISCUSSION THREAD

**Discussion by maphew** [prefix-dev/pixi#1859] [General]
Is it possible to use a conda environment on local disk and build a new pixi environment on top of that?

Pseudo code idea:
```
md test_foo
cd test_foo
pixi init
#e..g. os.environ['CONDA_PREFIX']
pixi add --from CONDA_PREFIX

# now add my new pixi layer of apps, without changing the source conda env
pixi add baz
pixi add bar
...etc
```

**Comment by maphew** [comment]
motivation: I'm trying to use an ArcGIS Pro arcpy python conda environment, but so far creating a pixi one from scratch hasn't worked for me. However the Pro conda environment works just fine. I'm trying to avoid using it's included Anaconda management tools to create new environments because it's so slow and painful.

**Reply by geoHeil** [reply]
At work I have a similar issue with mamba/conda-forge. What works for us there is to create a fresh environment without with the desired version of arcpy but then launch it inside the right folder so the licenses and DLLs of arcgis are discovered. Potentially this might work here as well

**Reply by baszalmstra** [reply]
You can export the environment using conda export and --import it on pixi init

**Reply by baszalmstra** [reply]
So also: https://prefix.dev/blog/pixi_a_fast_conda_alternative#switch-from-conda-to-pixi-in-one-step

**Reply by maphew** [reply]
> At work I have a similar issue with mamba/conda-forge. What works for us there is to create a fresh environment without with the desired version of arcpy but then launch it inside the right folder so the licenses and DLLs of arcgis are discovered. Potentially this might work here as well

When I try to create a new pixi env with only arcpy I get "failed to solve the conda requirements", but that's for another thread.

**Reply by maphew** [reply]
> When I try to create a new pixi env with only arcpy I get "failed to solve the conda requirements", but that's for another thread.
...opened at https://github.com/prefix-dev/pixi/issues/1876

**Comment by maphew** [comment]
To answer the thread, the recipe is:
```
conda env export -n foobaz > foobaz-environment.yml
pixi init --import foobaz-environment.yml
```
